crates/rpc-types-beacon/src/requests.rs

```diff
@@ -0,0 +1,148 @@
+                        }
+                        WITHDRAWAL_REQUEST_TYPE => {
+                            let list: Vec<WithdrawalRequest> = Vec::from_ssz_bytes(&request[1..])?;
+                            let size = list.len();
+                            if size > MAX_WITHDRAWAL_REQUESTS_PER_BLOCK {
+                                return Err(TryFromRequestsError::RequestPayloadSizeExceeded(
+                                    WITHDRAWAL_REQUEST_TYPE,
+                                    size,
+                                ));
+                            }
+                            acc.1.extend(list);
+                        }
+                        CONSOLIDATION_REQUEST_TYPE => {
+                            let list: Vec<ConsolidationRequest> =
+                                Vec::from_ssz_bytes(&request[1..])?;
+                            let size = list.len();
+                            if size > MAX_CONSOLIDATION_REQUESTS_PER_BLOCK {
+                                return Err(TryFromRequestsError::RequestPayloadSizeExceeded(
+                                    CONSOLIDATION_REQUEST_TYPE,
+                                    size,
+                                ));
+                            }
+                            acc.2.extend(list);
+                        }
+                        unknown => return Err(TryFromRequestsError::UnknownRequestType(unknown)),
+                    }
+                    Ok(acc)
+                },
+            )?;
+
+            Ok(Self { deposits, withdrawals, consolidations })
+        }
+    }
+
+    /// Errors possible converting a [Requests] to [ExecutionRequestsV4]
+    #[derive(Debug)]
+    pub enum TryFromRequestsError {
+        /// One of the Bytes is empty.
+        EmptyRequest,
+        /// Bytes prefix is not a known EIP-7685 request_type in Electra.
+        UnknownRequestType(u8),
+        /// Remaining bytes could not be decoded as SSZ requests_data.
+        SszDecodeError(DecodeError),
+        /// Requests of request_type exceeds Electra size limits
+        RequestPayloadSizeExceeded(u8, usize),
+    }
+
+    impl From<DecodeError> for TryFromRequestsError {
+        fn from(value: DecodeError) -> Self {
+            Self::SszDecodeError(value)
+        }
+    }
+
+    impl From<&ExecutionRequestsV4> for Requests {
+        fn from(val: &ExecutionRequestsV4) -> Self {
+            let deposit_bytes = val.deposits.as_ssz_bytes();
+            let withdrawals_bytes = val.withdrawals.as_ssz_bytes();
+            let consolidations_bytes = val.consolidations.as_ssz_bytes();
+
+            let mut requests = Self::default();
+            requests.push_request_with_type(DEPOSIT_REQUEST_TYPE, deposit_bytes);
+            requests.push_request_with_type(WITHDRAWAL_REQUEST_TYPE, withdrawals_bytes);
+            requests.push_request_with_type(CONSOLIDATION_REQUEST_TYPE, consolidations_bytes);
+            requests
+        }
```

Comment:
I'd prefer to have this as a native function of `ExecutionRequestsV4` as well

`ExecutionRequestsV4::to_requests(&self) {self.into()} `